ingestion/src/metadata/utils/dispatch.py

```diff
@@ -54,3 +56,53 @@ def inner(fn):
 
     Register = namedtuple("Register", ["add", "registry"])
     return Register(add, registry)
+
+
+def valuedispatch(func) -> Callable:
+    """Value dispatch for methods and functions
+
+    Args:
+        func (_type_): function to run
+
+    Returns:
+        Callable: wrapper
+    """
+
+    registry = {}
+
+    def _is_valid_dispatch(value):
+        return isinstance(value, str)
+
+    def dispatch(value: str) -> Callable:
+        try:
+            impl = registry[value]
+        except KeyError:
+            impl = registry[object]
+        return impl
+
+    def register(value, func=None) -> Callable:
+        if _is_valid_dispatch(value):
+            if func is None:
+                return lambda f: register(value, f)
```

Comment:
Not sure I follow this 🤔

Reply:
It simply allows you explicitly pass a function.